matRad_calcParticleDose.m

```diff
@@ -202,70 +228,110 @@
-                    continue;
-                end
-                
-                % adjust radDepth according to range shifter
-                currRadDepths = radDepths(currIx) + stf(i).ray(j).rangeShifter(k).eqThickness;
+                    % create offset vector to account for additional offsets modelled in the base data and a potential 
+                    % range shifter. In the following, we only perform dose calculation for voxels having a radiological depth
+                    % that is within the limits of the base data set (-> machine.data(i).dephts). By this means, we only allow  
+                    % interpolations in matRad_calcParticleDoseBixel() and avoid extrapolations.
+                    offsetRadDepth = machine.data(energyIx).offset - stf(i).ray(j).rangeShifter(k).eqThickness;
+
+                    % find depth depended lateral cut off
+                    if cutOffLevel >= 1
+                        currIx = radDepths <= machine.data(energyIx).depths(end) + offsetRadDepth;
+                    elseif cutOffLevel < 1 && cutOffLevel > 0
+                        % perform rough 2D clipping
+                        currIx = radDepths <= machine.data(energyIx).depths(end) + offsetRadDepth & ...
+                             currRadialDist_sq <= max(machine.data(energyIx).LatCutOff.CutOff.^2);
+
+                        % peform fine 2D clipping  
+                        if length(machine.data(energyIx).LatCutOff.CutOff) > 1
+                            currIx(currIx) = matRad_interp1((machine.data(energyIx).LatCutOff.depths + offsetRadDepth)',...
+                                (machine.data(energyIx).LatCutOff.CutOff.^2)', radDepths(currIx)) >= currRadialDist_sq(currIx);
+                        end
+                    else
+                        error('cutoff must be a value between 0 and 1')
+                    end
 
-                % calculate initial focus sigma
-                sigmaIni = matRad_interp1(machine.data(energyIx).initFocus.dist (stf(i).ray(j).focusIx(k),:)', ...
-                                             machine.data(energyIx).initFocus.sigma(stf(i).ray(j).focusIx(k),:)',stf(i).ray(j).SSD);
-                sigmaIni_sq = sigmaIni^2;
-                
-                % consider range shifter for protons if applicable
-                if stf(i).ray(j).rangeShifter(k).eqThickness > 0 && strcmp(pln.radiationMode,'protons')
+                    % empty bixels may happen during recalculation of error
+                    % scenarios -> skip to next bixel
+                    if ~any(currIx)
+                        continue;
+                    end
+
+                    % adjust radDepth according to range shifter
+                    currRadDepths = radDepths(currIx) + stf(i).ray(j).rangeShifter(k).eqThickness;
+
+                    % calculate initial focus sigma
+                    sigmaIni = matRad_interp1(machine.data(energyIx).initFocus.dist (stf(i).ray(j).focusIx(k),:)', ...
```

Comment:
wouldn't it make sense to also use the matRad_calcSigmaIni here? (I am also thinking about future developments where we may alternatively allow emittance parameterizations instead of interpolation).

Reply:
Yes, I think it would makes sense to do so. With this not being a fine sampling related change, should I still implement it with this PR?

Reply:
yes please implement it (and validate that at this point it really computes the same things